pygmt/xarray/accessor.py

```diff
@@ -137,9 +137,11 @@ def __init__(self, xarray_obj: xr.DataArray):
         # two columns of the shortened summary information of grdinfo.
         if (_source := self._obj.encoding.get("source")) and Path(_source).exists():
             with contextlib.suppress(ValueError):
-                self._registration, self._gtype = map(  # type: ignore[assignment]
+                _registration, _gtype = map(
                     int, grdinfo(_source, per_column="n").split()[-2:]
                 )
+                self._registration = GridRegistration(_registration)
+                self._gtype = GridType(_gtype)
```

Comment:
Maybe move the two lines out of the `contexlib.suppress` block, following 
https://github.com/GenericMappingTools/pygmt/pull/3448#issuecomment-2369952412
```suggestion
            self._registration = GridRegistration(_registration)
            self._gtype = GridType(_gtype)
```